meson.build

```diff
@@ -0,0 +1,112 @@
+project(
+  'afpfs-ng',
+  'c',
+  version: '0.8.2',
+  license: 'GPLv2',
+  default_options: ['warning_level=3', 'c_std=gnu99']
+)
+
+cc = meson.get_compiler('c')
+host_os = host_machine.system()
+
+uname = find_program('uname', required: false)
+
+uname_stdout = run_command(uname, '-a', check: false).stdout().strip()
+libsearch_dirs = []
+
+if host_os in ['dragonfly', 'freebsd', 'openbsd']
+    libsearch_dirs += '/usr/local/lib'
+endif
+
+if host_os == 'netbsd'
+    libsearch_dirs += '/usr/pkg/lib'
+endif
+
+if uname.found() and uname_stdout.to_lower().contains('omnios')
+    libsearch_dirs += '/opt/local/lib'
+endif
+
+cflags = ['-D_FILE_OFFSET_BITS=64']
+include_dirs = []
+
+if host_os in ['dragonfly', 'freebsd', 'openbsd']
+    include_dirs += ['/usr/local/include']
+    cflags += '-DLIBICONV_PLUG'
+elif host_os == 'netbsd'
+    include_dirs += ['/usr/pkg/include']
+elif host_os == 'omnios'
+    include_dirs += ['/opt/local/include']
+endif
+
+headers = [
+  'arpa/inet.h',
+  'fcntl.h',
+  'limits.h',
+  'netdb.h',
+  'readline/readline.h',
+  'stddef.h',
+  'stdlib.h',
+  'string.h',
+  'strings.h',
+  'sys/socket.h',
+  'sys/time.h',
+  'syslog.h',
+  'unistd.h',
+  'utime.h',
+  'iconv.h'
+]
+
+foreach h : headers
+  cc.has_header(h)
+endforeach
+
+functions = [
+  'bzero',
+  'gethostbyname',
+  'gettimeofday',
+  'inet_ntoa',
+  'memset',
+  'select',
+  'socket',
+  'strchr',
+  'strerror',
+  'strstr',
+  'strtol'
+]
+
+foreach f : functions
+  cc.has_function(f)
+endforeach
```

Comment:
You aren't doing anything with the return values for these checks. Regardless of whether they exist or not, all that happens is meson prints the word "YES" or "NO" and then continues building anyway.

If you want to compile different code depending on whether they exist, use the boolean return value of cc.has_function to set a -D macro of some sort (perhaps by feeding a configuration_data() dictionary to a `configure_file(output: 'config.h')`).

If you want to abort the build when these don't exist:
- compile time can do that for you anyway :D 
- Since meson 1.3.0, has_function can take `required: true` so configuration will fail if the function doesn't exist (this was a hole in meson until then -- has_header supported `required: true` since all the way back to 0.50.0)

Reply:
To me, the header and function checks are more for visibility when tinkering with cross platform compatibility. 99% of required symbols are captures by the library and dependency checks anyways.

The required headers are brought over from the autoconf macro verbatim so the first step would be to validate which ones are actually mandatory. I'll look into this in the future!

Reply:
And yes I agree that the compilation process will tell you without ambiguity when a mandatory header is missing. :)

Reply:
> To me, the header and function checks are more for visibility when tinkering with cross platform compatibility. 99% of required symbols are captured by the library and dependency checks anyways.

Maybe so, but remember each one requires meson to fork and run the compiler to perform the test. :) configuration will be faster if you do less unnecessary work...

autoconf projects have a habit of checking everything, mostly because autoconf encourages you to then write conditional fallback code based on `#define HAVE_FUNCNAME 1` in config.h, and it ends up being cargo-culted -- even for projects that never actually use those `HAVE_*` defines.

Reply:
That's a fair point. Maybe time to leave the autoconf cargo cult! It has been done.